examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -0,0 +1,290 @@
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+
+
+DECLARE_SHADER shader_1 KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array1 {
+  float[MAT_SIZE][MAT_SIZE] data1;
+};
+layout(std430, binding=1) buffer array1_mat_buf{
+  mat4x4 array1_mat[4][4];
+};
+void main() {
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    array1_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data1[row][col];
+}
+END
+
+COMPILE_SHADER shader_compiled_1 SHADER shader_1
+CREATE_PROGRAM compute_prog_1 SHADERS shader_compiled_1
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_1
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array1_mat_buf correct_array1_mat_buf
+
+DECLARE_SHADER shader_2 KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array2 {
+  float[MAT_SIZE][MAT_SIZE] data2;
+};
+layout(std430, binding=1) buffer array2_mat_buf{
+  mat4x4 array2_mat[4][4];
+};
+void main() {
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    array2_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data2[row][col];
+}
+END
+
+COMPILE_SHADER shader_compiled_2 SHADER shader_2
+CREATE_PROGRAM compute_prog_2 SHADERS shader_compiled_2
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_2
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array2_mat_buf correct_array2_mat_buf
+
+
+DECLARE_SHADER shader_3 KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array1 {
+  float[MAT_SIZE][MAT_SIZE] data1;
+};
+layout(std430, binding=1) buffer array2 {
+  float[MAT_SIZE][MAT_SIZE] data2;
+};
+layout(std430, binding=2) buffer array1_mat_buf{
+  mat4x4 array1_mat[4][4];
+};
+layout(std430, binding=3) buffer array2_mat_buf{
+  mat4x4 array2_mat[4][4];
+};
+layout(std430, binding=4) buffer result{
+  float[MAT_SIZE][MAT_SIZE] res;
+};
```

Comment:
The blocked matrix multiplication shader should produce a blocked matrix result, and you should use *another* shader to turn it back from blocked form into regular form.